src/main/java/net/neoforged/moddevgradle/internal/ModDevArtifactsWorkflow.java

```diff
@@ -123,6 +123,13 @@ public static ModDevArtifactsWorkflow create(Project project,
             task.getToolsJavaExecutable().set(javaToolchainService
                     .launcherFor(spec -> spec.getLanguageVersion().set(JavaLanguageVersion.of(versionCapabilities.javaVersion())))
                     .map(javaLauncher -> javaLauncher.getExecutablePath().getAsFile().getAbsolutePath()));
+            // NFRT itself needs to run with a newer version of the JDK to be able to compile with -release 25, for example
+            // It can however not run with Java 25 and compile Java 8 code while maintaining the same lambda naming.
+            if (versionCapabilities.javaVersion() > 21) {
+                task.getJavaExecutable().set(javaToolchainService
+                        .launcherFor(spec -> spec.getLanguageVersion().set(JavaLanguageVersion.of(versionCapabilities.javaVersion())))
+                        .map(javaLauncher -> javaLauncher.getExecutablePath().getAsFile().getAbsolutePath()));
+            }
```

Comment:
While this works, it seems more future proof to run javac in NFRT with the external tool JRE